### PR TITLE
fix(mongodb): Fix incorrect return type for calling Cursor.find with options

### PR DESCRIPTION
**Pull request by Celend**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## mongodb ([<kbd>unpkg</kbd>](https://unpkg.com/browse/mongodb@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'mongodb'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against eea1c960f58713b88f7293007af6b861ebcc2498
</p>

File: types/mongodb/test/cursor.ts [hunk 1]
```diff
@@ -34,11 +34,37 @@ async function run() {
...
+    tag: {
+      name: string;
+    };
+  }
+  const typedCollection = db.collection<TypedDoc>('test');
+  typedCollection.find({name: 'name'}, {}).map(x => x.tag);
+  typedCollection.find({'tag.name': 'name'}, {});
+  typedCollection.find({'tag.name': 'name'}, {projection: {name: 1, max: {$max: []}}})
+    .sort({score: {$meta: 'textScore'}})
+    .map(x => x.tag);
```
Comment (liorcode) [inline types/mongodb/test/cursor.ts:53]: Might also need to add a test for nested fields in projection:
```suggestion
  typedCollection.find({'tag.name': 'name'}, {projection: {'tag.name': 1, max: {$max: []}}})
    .sort({score: {$meta: 'textScore'}})
    .map(x => x.tag);
```

**Review by liorcode — APPROVED** [review]

**Comment by liorcode** [comment]
Can anyone please review and merge it? 🙏

**Review by CaselIT — APPROVED** [review]

**Comment by Celend** [comment]
Ready to merge

**Pull request merged**